File: TextWaveTests/MockObjects/MockUrlDownloader.m
//
//  MockUrlDownloader.m
//  LazyReader
//
//  Created by Nikola Sobadjiev on 5/29/14.
//  Copyright (c) 2014 Nikola Sobadjiev. All rights reserved.
//

#import "MockUrlDownloader.h"

@implementation MockUrlDownloader

- (void)createAndStartRequest
{
    return;
}

-(void) downloadResource
{
    [super downloadResource];
    if (self.shouldSimulateCompletion)
    {
        [self connectionDidFinishLoading:nil];
    }
    if (self.shouldSimulateFailure)
    {
        [self connection:nil 
        didFailWithError:nil];
    }
}

@end

File: TextWaveTests/MockObjects/MockUrlDownloader.swift
//
//  MockUrlDownloader.swift
//  LazyReader
//
//  Created by Nikola Sobadjiev on 7/18/14.
//  Copyright (c) 2014 Nikola Sobadjiev. All rights reserved.
//

import Foundation

class MockUrlDownloader : UrlDownloader {
    var shouldSimulateCompletion: Bool = false
    var shouldSimulateFailure: Bool = false
    
    override func createAndStartRequest() {
        
    }
    
    override func downloadResource() {
        if (self.shouldSimulateCompletion) {
            self.connectionDidFinishLoading(nil)
        }
        else if (self.shouldSimulateFailure) {
            self.connection(nil, didFailWithError: nil)
        }
    }
}
